tests/test_content_streams.py

```diff
@@ -189,6 +189,21 @@ def hello_world():
     )
 
 
+@pytest.mark.asyncio
+async def test_empty_request():
+    stream = encode(data={}, files={})
+    sync_content = b"".join([part for part in stream])
+    async_content = b"".join([part async for part in stream])
+
+    assert stream.can_replay()
+    assert stream.get_headers() == {
+        "Content-Length": "0",
+        "Content-Type": "application/x-www-form-urlencoded",
```

Comment:
(And so to match Requests there shouldn't be any `Content-Type` in the request. :-))

Reply:
I found that ByteStream.get_headers doesn't return Content-Length header with an empty body. Should I fix it?
https://github.com/encode/httpx/blob/43ec09c3cbb089edd2e83ccc717fb95beef0c189/httpx/_content_streams.py#L73-L75

Reply:
Because the body is `bytes` (`Union[str, bytes]` transformed in `__init__` to `bytes`) this check looks redundant 🤔

Reply:
Looking at [RFC 7320](https://tools.ietf.org/html/rfc7230#section-3.3.2), it looks like for POST requests including a `Content-Length` always is sort of recommended, yes:

> A user agent SHOULD send a Content-Length in a request message when
   no Transfer-Encoding is sent and the request method defines a meaning
   for an enclosed payload body. **For example, a Content-Length header
   field is normally sent in a POST request even when the value is 0
   (indicating an empty payload body).**

So given that we only use `encode()` for multipart POST requests, I think it would make sense for us to do it. But let's treat that as a separate PR maybe? 😄

Reply:
✅ Done!